stdlib/src/builtin/builtin_slice.mojo

```diff
@@ -75,35 +62,33 @@ struct Slice(Sized, Stringable, EqualityComparable):
         self.step = 1
 
     @always_inline("nodebug")
-    fn __init__[
-        T0: AnyRegType, T1: AnyRegType, T2: AnyRegType
-    ](inout self, start: T0, end: T1, step: T2):
+    fn __init__(
+        inout self,
+        start: OptionalReg[Int],
+        end: OptionalReg[Int],
+        step: OptionalReg[Int],
+    ):
         """Construct slice given the start, end and step values.
 
-        Parameters:
-            T0: Type of the start value.
-            T1: Type of the end value.
-            T2: Type of the step value.
-
         Args:
             start: The start value.
             end: The end value.
             step: The step value.
         """
-        self.start = _default_or(start, 0)
-        self.end = _default_or(end, _int_max_value())
-        self.step = _default_or(step, 1)
+        self.start = start
+        self.end = end
+        self.step = step.value() if step else 1
```

Comment:
BTW, no need to fix this in this patch if we don't comply with this; followup is fine.

Reply:
Thanks for pointing that out, currently that results in unexpected behaviour (apparently we also don't throw an error when giving `range` a 0 step either).

That might be better as a follow up as well since having to mark all slice getters with `raises` feels like a pretty significant commitment even if it is in the name of being python compliant.

Reply:
Yeah, let's not raise from these, but try to do either `debug_assert` or check and `abort`.

Reply:
Ok in that case I've just added a check and abort in the constructor